### PR TITLE
Split GitHub-Actions tests

### PR DESCRIPTION
**Pull request by Or-Geva**
- [ ] All [tests](https://github.com/jfrog/jfrog-cli#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [ ] All [static analysis checks](https://github.com/jfrog/jfrog-cli/actions/workflows/analysis.yml) passed.
- [ ] This pull request is on the dev branch.
- [ ] I used gofmt for formatting the code before submitting the pull request.
-----
* Move all the tests section from README.md to TESTS.md 
* Split GitHub-Actions tests into multiple suites

### CODE REVIEW

File: .github/workflows/artifactoryTests.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  Artifactory-Tests:
+    if: contains(github.event.pull_request.labels.*.name, 'safe to test') || github.event_name == 'push'
+    name: ${{ matrix.suite }} (${{ matrix.os }})
+    strategy:
+      fail-fast: false
+      matrix:
+        suite: [artifactory, artifactoryProject]
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
```
Comment (yahavi) [inline .github/workflows/artifactoryTests.yml:22]: I couldn't find checkout in this workflow

File: .github/workflows/distributionTests.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  Distribution-Tests:
+    if: contains(github.event.pull_request.labels.*.name, 'safe to test') || github.event_name == 'push'
+    name: ${{ matrix.suite }} (${{ matrix.os }})
+    strategy:
+      fail-fast: false
+      matrix:
+        suite: [distribution]
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
```
Comment (yahavi) [inline .github/workflows/distributionTests.yml:22]: I couldn't find checkout in this workflow

File: .github/workflows/dockerTests.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+
```
Comment (yahavi) [inline .github/workflows/dockerTests.yml:1]:
```suggestion
```

File: .github/workflows/goTests.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  group: ${{ github.event.pull_request.number || github.sha }}go
+  cancel-in-progress: true
+jobs:
+  GO-tests:
+      # Go modules doesn't allow passing credentials to a private registry using an HTTP URL. Therefore, the Go tests run against a remote Artifactory server.
+    if: contains(github.event.pull_request.labels.*.name, 'safe to test') || github.event_name == 'push'
+    name: GO (${{ matrix.os }})
+    defaults:
+      run:
+        shell: bash
```
Comment (yahavi) [inline .github/workflows/goTests.yml:20]: Is it needed?
```suggestion
```

File: .github/workflows/goTests.yml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    types: [labeled]
+# Ensures that only the latest commit is running for each PR at a time.
+# Ignores this rule for push events.
+concurrency:
+  group: ${{ github.event.pull_request.number || github.sha }}go
+  cancel-in-progress: true
+jobs:
+  GO-tests:
+      # Go modules doesn't allow passing credentials to a private registry using an HTTP URL. Therefore, the Go tests run against a remote Artifactory server.
+    if: contains(github.event.pull_request.labels.*.name, 'safe to test') || github.event_name == 'push'
```
Comment (yahavi) [inline .github/workflows/goTests.yml:16]:
```suggestion
    # Go modules doesn't allow passing credentials to a private registry using an HTTP URL. Therefore, the Go tests run against a remote Artifactory server.
    if: contains(github.event.pull_request.labels.*.name, 'safe to test') || github.event_name == 'push'
```

File: .github/workflows/gradleTests.yml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  group: ${{ github.event.pull_request.number || github.sha }}gradle
+  cancel-in-progress: true
+jobs:
+  Gradle-Tests:
+    if: contains(github.event.pull_request.labels.*.name, 'safe to test') || github.event_name == 'push'
+    name: ${{ matrix.suite }} (${{ matrix.os }})
+    strategy:
+      fail-fast: false
+      matrix:
+        suite: [gradle]
```
Comment (yahavi) [inline .github/workflows/gradleTests.yml:19]:
```suggestion
```

File: .github/workflows/gradleTests.yml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
+    env:
+      GRADLE_OPTS: -Dorg.gradle.daemon=false
+    steps:
```
Comment (yahavi) [inline .github/workflows/gradleTests.yml:24]: I couldn't find checkout in this workflow

File: .github/workflows/gradleTests.yml [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+      - name: Setup Artifactory
+        run: |
+          go install github.com/jfrog/jfrog-testing-infra/local-rt-setup@latest
+          ~/go/bin/local-rt-setup
+        env:
+          RTLIC: ${{secrets.RTLIC}}
+          GOPROXY: direct
+
+      - name: Run Gradle tests
+        run: go test -v github.com/jfrog/jfrog-cli --timeout 0 --test.${{ matrix.suite }}=true --ci.runId=${{ runner.os }}-${{ matrix.suite }}
```
Comment (yahavi) [inline .github/workflows/gradleTests.yml:46]:
```suggestion
        run: go test -v github.com/jfrog/jfrog-cli --timeout 0 --test.gradle --ci.runId=${{ runner.os }}-gradle
```

File: .github/workflows/accessTests.yml [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+  group: ${{ github.event.pull_request.number || github.sha }}access
+  cancel-in-progress: true
+jobs:
+  CLI-Tests:
+    if: contains(github.event.pull_request.labels.*.name, 'safe to test') || github.event_name == 'push'
+    name: Access (${{ matrix.os }})
+    strategy:
+      fail-fast: false
+      matrix:
+        suite: [access]
```
Comment (yahavi) [inline .github/workflows/accessTests.yml:19]:
```suggestion
```

File: .github/workflows/accessTests.yml [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+        with:
+          go-version: 1.17.x
+
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Run Access tests
+        run: go test -v github.com/jfrog/jfrog-cli --timeout 0 --test.${{ matrix.suite }}=true --jfrog.url=${{ secrets.PLATFORM_URL }} --jfrog.adminToken=${{ secrets.PLATFORM_ADMIN_TOKEN }} --jfrog.user=${{ secrets.PLATFORM_USER }} --ci.runId=${{ runner.os }}-${{ matrix.suite }}
```
Comment (yahavi) [inline .github/workflows/accessTests.yml:34]:
```suggestion
        run: go test -v github.com/jfrog/jfrog-cli --timeout 0 --test.access --jfrog.url=${{ secrets.PLATFORM_URL }} --jfrog.adminToken=${{ secrets.PLATFORM_ADMIN_TOKEN }} --jfrog.user=${{ secrets.PLATFORM_USER }} --ci.runId=${{ runner.os }}-access
```

File: .github/workflows/distributionTests.yml [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+  group: ${{ github.event.pull_request.number || github.sha }}distribution
+  cancel-in-progress: true
+jobs:
+  Distribution-Tests:
+    if: contains(github.event.pull_request.labels.*.name, 'safe to test') || github.event_name == 'push'
+    name: ${{ matrix.suite }} (${{ matrix.os }})
+    strategy:
+      fail-fast: false
+      matrix:
+        suite: [distribution]
```
Comment (yahavi) [inline .github/workflows/distributionTests.yml:19]:
```suggestion
```

File: .github/workflows/distributionTests.yml [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.17.x
+
+      - name: Run Distribution tests
+        run: go test -v github.com/jfrog/jfrog-cli --timeout 0 --test.${{ matrix.suite }}=true --jfrog.url=${{ secrets.PLATFORM_URL }} --jfrog.adminToken=${{ secrets.PLATFORM_ADMIN_TOKEN }} --jfrog.user=${{ secrets.PLATFORM_USER }} --ci.runId=${{ runner.os }}-${{ matrix.suite }}
```
Comment (yahavi) [inline .github/workflows/distributionTests.yml:29]:
```suggestion
        run: go test -v github.com/jfrog/jfrog-cli --timeout 0 --test.distribution --jfrog.url=${{ secrets.PLATFORM_URL }} --jfrog.adminToken=${{ secrets.PLATFORM_ADMIN_TOKEN }} --jfrog.user=${{ secrets.PLATFORM_USER }} --ci.runId=${{ runner.os }}-distribution
```

File: .github/workflows/dockerTests.yml [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+      - name: Go Cache
+        uses: actions/cache@v3
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: ${{ runner.os }}-go-
+
+      - name: Run Docker tests
+        run: go test -v -timeout 0 --test.docker=true --jfrog.url=${{ secrets.PLATFORM_URL }} --jfrog.adminToken=${{ secrets.PLATFORM_ADMIN_TOKEN }} --rt.dockerRepoDomain=${{ secrets.CONTAINER_REGISTRY }} --rt.dockerVirtualRepo=${{ secrets.DOCKER_VIRTUAL }} --rt.dockerLocalRepo=${{ secrets.DOCKER_LOCAL }} --rt.dockerRemoteRepo=${{ secrets.DOCKER_REMOTE }} --rt.dockerPromoteLocalRepo=${{ secrets.DOCKER_PROMOTE_LOCAL }}
```
Comment (yahavi) [inline .github/workflows/dockerTests.yml:36]:
```suggestion
        run: go test -v -timeout 0 --test.docker --jfrog.url=${{ secrets.PLATFORM_URL }} --jfrog.adminToken=${{ secrets.PLATFORM_ADMIN_TOKEN }} --rt.dockerRepoDomain=${{ secrets.CONTAINER_REGISTRY }} --rt.dockerVirtualRepo=${{ secrets.DOCKER_VIRTUAL }} --rt.dockerLocalRepo=${{ secrets.DOCKER_LOCAL }} --rt.dockerRemoteRepo=${{ secrets.DOCKER_REMOTE }} --rt.dockerPromoteLocalRepo=${{ secrets.DOCKER_PROMOTE_LOCAL }}
```

File: .github/workflows/lintTest.yml [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Go Cache
+        uses: actions/cache@v3
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: ${{ runner.os }}-go-
+
+      - name: Run lint vet
```
Comment (yahavi) [inline .github/workflows/lintTest.yml:39]:
```suggestion
      - name: Run Go vet
```

File: .github/workflows/mavenTests.yml [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+  group: ${{ github.event.pull_request.number || github.sha }}mvn
+  cancel-in-progress: true
+jobs:
+  mvn-Tests:
+    if: contains(github.event.pull_request.labels.*.name, 'safe to test') || github.event_name == 'push'
+    name: ${{ matrix.suite }} (${{ matrix.os }})
+    strategy:
+      fail-fast: false
+      matrix:
+        suite: [maven]
```
Comment (yahavi) [inline .github/workflows/mavenTests.yml:19]:
```suggestion
```

File: .github/workflows/mavenTests.yml [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+      - name: Setup Artifactory
+        run: |
+          go install github.com/jfrog/jfrog-testing-infra/local-rt-setup@latest
+          ~/go/bin/local-rt-setup
+        env:
+          RTLIC: ${{secrets.RTLIC}}
+          GOPROXY: direct
+
+      - name: Run mvn tests
+        run: go test -v github.com/jfrog/jfrog-cli --timeout 0 --test.${{ matrix.suite }}=true --ci.runId=${{ runner.os }}-${{ matrix.suite }}
```
Comment (yahavi) [inline .github/workflows/mavenTests.yml:44]:
```suggestion
        run: go test -v github.com/jfrog/jfrog-cli --timeout 0 --test.maven --ci.runId=${{ runner.os }}-maven
```

File: .github/workflows/npmTests.yml [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+  group: ${{ github.event.pull_request.number || github.sha }}npm
+  cancel-in-progress: true
+jobs:
+  npm-Tests:
+    if: contains(github.event.pull_request.labels.*.name, 'safe to test') || github.event_name == 'push'
+    name: ${{ matrix.suite }} (${{ matrix.os }})
+    strategy:
+      fail-fast: false
+      matrix:
+        suite: [npm]
```
Comment (yahavi) [inline .github/workflows/npmTests.yml:19]:
```suggestion
```

File: .github/workflows/npmTests.yml [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+      - name: Setup Artifactory
+        run: |
+          go install github.com/jfrog/jfrog-testing-infra/local-rt-setup@latest
+          ~/go/bin/local-rt-setup
+        env:
+          RTLIC: ${{secrets.RTLIC}}
+          GOPROXY: direct
+
+      - name: Run npm tests
+        run: go test -v github.com/jfrog/jfrog-cli --timeout 0 --test.${{ matrix.suite }}=true --ci.runId=${{ runner.os }}-${{ matrix.suite }}
```
Comment (yahavi) [inline .github/workflows/npmTests.yml:54]:
```suggestion
        run: go test -v github.com/jfrog/jfrog-cli --timeout 0 --test.npm --ci.runId=${{ runner.os }}-npm
```

File: .github/workflows/nugetTests.yml [hunk 19]
```diff
@@ -0,0 +1,61 @@
...
+  group: ${{ github.event.pull_request.number || github.sha }}nuget
+  cancel-in-progress: true
+jobs:
+  NuGet-Tests:
+    if: contains(github.event.pull_request.labels.*.name, 'safe to test') || github.event_name == 'push'
+    name: ${{ matrix.suite }} (${{ matrix.os }})
+    strategy:
+      fail-fast: false
+      matrix:
+        suite: [nuget]
```
Comment (yahavi) [inline .github/workflows/nugetTests.yml:19]:
```suggestion
```

File: .github/workflows/nugetTests.yml [hunk 20]
```diff
@@ -0,0 +1,61 @@
...
+      - name: Setup Artifactory
+        run: |
+          go install github.com/jfrog/jfrog-testing-infra/local-rt-setup@latest
+          ~/go/bin/local-rt-setup
+        env:
+          RTLIC: ${{secrets.RTLIC}}
+          GOPROXY: direct
+
+      - name: Run nuget tests
+        run: go test -v github.com/jfrog/jfrog-cli --timeout 0 --test.${{ matrix.suite }}=true --ci.runId=${{ runner.os }}-${{ matrix.suite }}
```
Comment (yahavi) [inline .github/workflows/nugetTests.yml:61]:
```suggestion
        run: go test -v github.com/jfrog/jfrog-cli --timeout 0 --test.nuget --ci.runId=${{ runner.os }}-nuget
```

File: .github/workflows/goTests.yml [hunk 21]
```diff
@@ -0,0 +1,45 @@
+
+name: GO Tests
```
Comment (yahavi) [inline .github/workflows/goTests.yml:2]:
```suggestion
name: Go Tests
```

File: .github/workflows/lintTest.yml [hunk 22]
```diff
@@ -0,0 +1,40 @@
+name: Lint Tests
```
Comment (yahavi) [inline .github/workflows/lintTest.yml:1]:
```suggestion
name: Lint
```

Comment (yahavi) [inline .github/workflows/lintTest.yml:1]: I suggest moving the whole `go lint` to the analysis.yml

File: .github/workflows/mavenTests.yml [hunk 23]
```diff
@@ -0,0 +1,44 @@
+name: mvn Tests
```
Comment (yahavi) [inline .github/workflows/mavenTests.yml:1]:
```suggestion
name: Maven Tests
```

File: .github/workflows/nugetTests.yml [hunk 24]
```diff
@@ -0,0 +1,61 @@
+name: Tests
```
Comment (yahavi) [inline .github/workflows/nugetTests.yml:1]:
```suggestion
name: NuGet Tests
```

File: .github/workflows/pipTests.yml [hunk 25]
```diff
@@ -0,0 +1,58 @@
+name: pip Tests
```
Comment (yahavi) [inline .github/workflows/pipTests.yml:1]:
```suggestion
name: Pip Tests
```

File: .github/workflows/pipTests.yml [hunk 26]
```diff
@@ -0,0 +1,58 @@
...
+  group: ${{ github.event.pull_request.number || github.sha }}pip
+  cancel-in-progress: true
+jobs:
+  Pip-Tests:
+    if: contains(github.event.pull_request.labels.*.name, 'safe to test') || github.event_name == 'push'
+    name: ${{ matrix.suite }} (${{ matrix.os }})
+    strategy:
+      fail-fast: false
+      matrix:
+        suite: [pip]
```
Comment (yahavi) [inline .github/workflows/pipTests.yml:19]:
```suggestion
```

File: .github/workflows/pipTests.yml [hunk 27]
```diff
@@ -0,0 +1,58 @@
...
+        with:
+          go-version: 1.17.x
+
+      - name: Setup Python3
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.x"
+
+      - name: Setup Pipenv
+        if: ${{ matrix.suite == 'pipenv' || matrix.suite == 'xray' }}
```
Comment (yahavi) [inline .github/workflows/pipTests.yml:34]:
```suggestion
```

File: .github/workflows/pipTests.yml [hunk 28]
```diff
@@ -0,0 +1,58 @@
...
+      - name: Setup Artifactory
+        run: |
+          go install github.com/jfrog/jfrog-testing-infra/local-rt-setup@latest
+          ~/go/bin/local-rt-setup
+        env:
+          RTLIC: ${{secrets.RTLIC}}
+          GOPROXY: direct
+
+      - name: Run pip tests
+        run: go test -v github.com/jfrog/jfrog-cli --timeout 0 --test.${{ matrix.suite }}=true --ci.runId=${{ runner.os }}-${{ matrix.suite }}
```
Comment (yahavi) [inline .github/workflows/pipTests.yml:58]:
```suggestion
        run: go test -v github.com/jfrog/jfrog-cli --timeout 0 --test.pip --ci.runId=${{ runner.os }}-pip
```

File: .github/workflows/pipTests.yml [hunk 29]
```diff
@@ -0,0 +1,58 @@
...
+
+      - name: Setup Artifactory
+        run: |
+          go install github.com/jfrog/jfrog-testing-infra/local-rt-setup@latest
+          ~/go/bin/local-rt-setup
+        env:
+          RTLIC: ${{secrets.RTLIC}}
+          GOPROXY: direct
+
+      - name: Run pip tests
```
Comment (yahavi) [inline .github/workflows/pipTests.yml:57]:
```suggestion
      - name: Run Pip tests
```

File: .github/workflows/nugetTests.yml [hunk 30]
```diff
@@ -0,0 +1,61 @@
...
+
+      - name: Setup Artifactory
+        run: |
+          go install github.com/jfrog/jfrog-testing-infra/local-rt-setup@latest
+          ~/go/bin/local-rt-setup
+        env:
+          RTLIC: ${{secrets.RTLIC}}
+          GOPROXY: direct
+
+      - name: Run nuget tests
```
Comment (yahavi) [inline .github/workflows/nugetTests.yml:60]:
```suggestion
      - name: Run NuGet tests
```

File: .github/workflows/pluginsTests.yml [hunk 31]
```diff
@@ -0,0 +1,49 @@
...
+  group: ${{ github.event.pull_request.number || github.sha }}plugins
+  cancel-in-progress: true
+jobs:
+  Plugins-Tests:
+    if: contains(github.event.pull_request.labels.*.name, 'safe to test') || github.event_name == 'push'
+    name: ${{ matrix.suite }} (${{ matrix.os }})
+    strategy:
+      fail-fast: false
+      matrix:
+        suite: [plugins]
```
Comment (yahavi) [inline .github/workflows/pluginsTests.yml:19]:
```suggestion
```

File: .github/workflows/pluginsTests.yml [hunk 32]
```diff
@@ -0,0 +1,49 @@
...
+      - name: Setup Artifactory
+        run: |
+          go install github.com/jfrog/jfrog-testing-infra/local-rt-setup@latest
+          ~/go/bin/local-rt-setup
+        env:
+          RTLIC: ${{secrets.RTLIC}}
+          GOPROXY: direct
+
+      - name: Run plugins tests
+        run: go test -v github.com/jfrog/jfrog-cli --timeout 0 --test.${{ matrix.suite }}=true --ci.runId=${{ runner.os }}-${{ matrix.suite }}
```
Comment (yahavi) [inline .github/workflows/pluginsTests.yml:49]:
```suggestion
        run: go test -v github.com/jfrog/jfrog-cli --timeout 0 --test.plugins --ci.runId=${{ runner.os }}-plugins
```

File: .github/workflows/scriptTests.yml [hunk 33]
```diff
@@ -0,0 +1,37 @@
+
+name: Scripts Tests
```
Comment (yahavi) [inline .github/workflows/scriptTests.yml:2]:
```suggestion
name: Scripts Tests
```

File: .github/workflows/scriptTests.yml [hunk 34]
```diff
@@ -0,0 +1,37 @@
...
+  pull_request_target:
+    types: [labeled]
+# Ensures that only the latest commit is running for each PR at a time.
+# Ignores this rule for push events.
+concurrency:
+  group: ${{ github.event.pull_request.number || github.sha }}script
+  cancel-in-progress: true
+jobs:
+  Scripts-tests:
+    if: contains(github.event.pull_request.labels.*.name, 'safe to test') || github.event_name == 'push'
```
Comment (yahavi) [inline .github/workflows/scriptTests.yml:15]: This workflow does not use credentials and therefore "safe to test" is not needed
```suggestion
```

File: .github/workflows/xrayTests.yml [hunk 35]
```diff
@@ -0,0 +1,70 @@
...
+  group: ${{ github.event.pull_request.number || github.sha }}xray
+  cancel-in-progress: true
+jobs:
+  CLI-Tests:
+    if: contains(github.event.pull_request.labels.*.name, 'safe to test') || github.event_name == 'push'
+    name: ${{ matrix.suite }} (${{ matrix.os }})
+    strategy:
+      fail-fast: false
+      matrix:
+        suite: [xray]
```
Comment (yahavi) [inline .github/workflows/xrayTests.yml:19]:
```suggestion
```

File: .github/workflows/xrayTests.yml [hunk 36]
```diff
@@ -0,0 +1,70 @@
...
+        with:
+          node-version: "16"
+        if: ${{ matrix.suite == 'npm' || matrix.suite == 'xray' }}
+
+      - name: Install Java
+        uses: actions/setup-java@v3
+        with:
+          java-version: "11"
+          distribution: "adopt"
+        if: ${{ matrix.suite == 'maven' || matrix.suite == 'gradle' || matrix.suite == 'xray' }}
```
Comment (yahavi) [inline .github/workflows/xrayTests.yml:41]:
```suggestion
```

File: .github/workflows/xrayTests.yml [hunk 37]
```diff
@@ -0,0 +1,70 @@
...
+      - name: Install Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.17.x
+
+      - name: Install npm
+        uses: actions/setup-node@v3
+        with:
+          node-version: "16"
+        if: ${{ matrix.suite == 'npm' || matrix.suite == 'xray' }}
```
Comment (yahavi) [inline .github/workflows/xrayTests.yml:34]:
```suggestion
```

File: .github/workflows/xrayTests.yml [hunk 38]
```diff
@@ -0,0 +1,70 @@
...
+        with:
+          java-version: "11"
+          distribution: "adopt"
+        if: ${{ matrix.suite == 'maven' || matrix.suite == 'gradle' || matrix.suite == 'xray' }}
+
+      - name: Install NuGet
+        uses: nuget/setup-nuget@v1
+        with:
+          nuget-version: 5.x
+        if: ${{ matrix.suite == 'nuget' }}
```
Comment (yahavi) [inline .github/workflows/xrayTests.yml:47]:
```suggestion
```

File: .github/workflows/xrayTests.yml [hunk 39]
```diff
@@ -0,0 +1,70 @@
...
+
+      - name: Install dotnet
+        uses: actions/setup-dotnet@v2
+        with:
+          dotnet-version: "3.x"
+        if: ${{ matrix.suite == 'nuget' }}
```
Comment (yahavi) [inline .github/workflows/xrayTests.yml:53]:
```suggestion
```

File: .github/workflows/xrayTests.yml [hunk 40]
```diff
@@ -0,0 +1,70 @@
...
+
+      - name: Setup Python3
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.x"
+
+      - name: Setup Pipenv
+        if: ${{ matrix.suite == 'pipenv' || matrix.suite == 'xray' }}
```
Comment (yahavi) [inline .github/workflows/xrayTests.yml:61]:
```suggestion
```

File: .github/workflows/xrayTests.yml [hunk 41]
```diff
@@ -0,0 +1,70 @@
...
+        run: pip3 install pipenv
+
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Run Distribution / Xray / Access tests
+        run: go test -v github.com/jfrog/jfrog-cli --timeout 0 --test.${{ matrix.suite }}=true --jfrog.url=${{ secrets.PLATFORM_URL }} --jfrog.adminToken=${{ secrets.PLATFORM_ADMIN_TOKEN }} --jfrog.user=${{ secrets.PLATFORM_USER }} --ci.runId=${{ runner.os }}-${{ matrix.suite }}
```
Comment (yahavi) [inline .github/workflows/xrayTests.yml:70]:
```suggestion
        run: go test -v github.com/jfrog/jfrog-cli --timeout 0 --test.xray --jfrog.url=${{ secrets.PLATFORM_URL }} --jfrog.adminToken=${{ secrets.PLATFORM_ADMIN_TOKEN }} --jfrog.user=${{ secrets.PLATFORM_USER }} --ci.runId=${{ runner.os }}-xray
```

File: .github/workflows/xrayTests.yml [hunk 42]
```diff
@@ -0,0 +1,70 @@
...
+      - name: Setup Pipenv
+        if: ${{ matrix.suite == 'pipenv' || matrix.suite == 'xray' }}
+        run: pip3 install pipenv
+
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Run Distribution / Xray / Access tests
```
Comment (yahavi) [inline .github/workflows/xrayTests.yml:69]:
```suggestion
      - name: Run Xray tests
```

File: TESTS.md [hunk 43]
```diff
@@ -0,0 +1,413 @@
...
+```
+
+The flags are:
+
+| Flag                | Description                                                                                     |
+| ------------------- | ----------------------------------------------------------------------------------------------- |
+| `-jfrog.url`        | [Default: <http://localhost:8081>] JFrog platform URL.                                          |
+| `-jfrog.user`       | [Default: admin] JFrog platform username.                                                       |
+| `-jfrog.password`   | [Default: password] JFrog platform password.                                                    |
+| `-jfrog.adminToken` | JFrog platform admin token.                                                                     |
```
Comment (yahavi) [inline TESTS.md:234]:
```suggestion
| `-jfrog.adminToken` | [Optional] JFrog platform admin token.                                                           |
```

File: .github/workflows/accessTests.yml [hunk 44]
```diff
@@ -0,0 +1,34 @@
...
+  # Triggers the workflow on labeled PRs only.
+  pull_request_target:
+    types: [labeled]
+# Ensures that only the latest commit is running for each PR at a time.
+# Ignores this rule for push events.
+concurrency:
+  group: ${{ github.event.pull_request.number || github.sha }}access
+  cancel-in-progress: true
+jobs:
+  CLI-Tests:
```
Comment (yahavi) [inline .github/workflows/accessTests.yml:13]:
```suggestion
  Access-Tests:
```

**Review by yahavi — CHANGES_REQUESTED** [review]
Looks good!
Please consider the following guidelines:
1. Since there is a large number of code repetitions, I suggest keeping things as short and simple as possible. If possible in GitHub Actions, please consider reusing steps.
2. I found some formatting issues - please make sure to reformat the code in all files.
3. A boolean flag does not need `=true` (see in my comments).
4. I think a suite of 1 component is not needed (see in my comments).

File: README.md [hunk 45]
```diff
@@ -9,19 +9,12 @@
...
-|   v1   |                                                                                             [![JFrog CLI Tests](https://github.com/jfrog/jfrog-cli/actions/workflows/tests.yml/badge.svg?branch=v1)](https://github.com/jfrog/jfrog-cli/actions/workflows/tests.yml)                                                                                             |
-| dev-v1 |                                                                                           [![JFrog CLI Tests](https://github.com/jfrog/jfrog-cli/actions/workflows/tests.yml/badge.svg?branch=dev-v1)](https://github.com/jfrog/jfrog-cli/actions/workflows/tests.yml)                                                                                           |
-
 # Table of Contents
 
 - [Overview](#overview)
 - [Download and Installation](#download-and-installation)
 - [Building the Executable](#building-the-executable)
-- [Tests](#tests)
+- [Tests](./TESTS.md)
```
Comment (yahavi) [inline README.md:17]: The tests are not in this file and therefore it is not reasonable to add them to the table of contents.
```suggestion
```

As an alternative, I suggest adding a "tests" or "code status" badge leading to /TESTS.md.

**Comment by yahavi** [comment]
Let's also add the new Transfer files integration test to the list of badges

File: README.md (hunk 45 above)
Comment (Or-Geva) [reply README.md:17]: The TEST.md file has been moved back to the readme

**Pull request merged**